dadi/frontend/components/FieldImage/FieldImage.jsx

```diff
@@ -1,12 +1,100 @@
 'use strict'
 
 import { h, Component } from 'preact'
+import 'unfetch'
+
+import Label from 'components/Label/Label'
 
 export default class FieldImage extends Component { 
 
+  constructor (props) {
+    super(props)
+
+    this.state.fileName = ''
+    this.state.signedUrl = ''
+    this.state.file = null
+    this.state.error = ''
+  }
+
   render() {
+    const { config } = this.props
+
     return (
-      <div>Image</div>
+      <Label label="Image">
+        <p>{this.state.signedUrl}</p>
+        {config && config.FieldImage && (
+          <input 
+            type="file" 
+            accept={ config.FieldImage.accept } 
+            onChange={this.handleFileChange.bind(this)}
+          />
+        )}
+        <pre>{this.state.error}</pre>
+      </Label>
     )
   }
+
+  uploadToS3 () {
+
+    let body = new FormData()
+    body.append('file', this.state.file)
+
+    let options = {
+      method: 'PUT',
+      body: this.state.file,
+      headers: {
+        'Content-Type': this.state.file.type,
+        'x-amz-acl': 'public-read'
+      }
+    }
+
+
+    fetch(this.state.signedUrl, options).then(res => {
+      res.text().then(resp => {
+        console.log(resp)
+      })
+    }).catch(err => {
+      console.log("ERR", err)
+    })
+  }
+
+  handleFileChange (event) {
+    event.preventDefault()
+
+    const { config } = this.props
+
+    this.setState({
+      fileName: event.target.files[0].name,
+      file: event.target.files[0]
+    })
+
+    if (config.FieldImage.s3.enabled) {
+
```

Comment:
minor: we've not been using blank lines at the beginning of function declarations or any other blocks.